Add vlan and mtu parameters to OVNOverlayNetworkAttachmentDefinition

Anatw opened the pull request:
Needed for the localnet feature.

##### Short description:

##### More details:

##### What this PR does / why we need it:

##### Which issue(s) this PR fixes:

##### Special notes for reviewer:

##### Bug:

redhat-qe-bot1 commented:
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox
 * `/retest python-module-install`: Retest python-module-install

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

Anatw commented:
/verified
https://main-jenkins-csb-cnvqe.apps.ocp-c1.prod.psi.redhat.com/job/cnv-tests-runner-containerized/4031/

redhat-qe-bot2 commented:
Label job is not a predefined one, will not be added / removed.
Available labels:

 * hold
 * verified
 * wip
 * lgtm

redhat-qe-bot2 commented:
Label cnv-tests-runner-containerized is not a predefined one, will not be added / removed.
Available labels:

 * hold
 * verified
 * wip
 * lgtm

redhat-qe-bot2 commented:
Label 4031 is not a predefined one, will not be added / removed.
Available labels:

 * hold
 * verified
 * wip
 * lgtm

Anatw commented:
/verified

myakove commented on ocp_resources/network_attachment_definition.py:
```diff
@@ -211,8 +213,12 @@ def __init__(
-                file.
+            network_name (str): The name of the network, used to connect resources
+                created in different namespaces to the same network. This field is
+                mandatory when not using yaml file.
```
not needed `This field is mandatory when not using yaml file`
you don;t have optional in the docstring.

Anatw replied:
ack

Anatw commented:
https://main-jenkins-csb-cnvqe.apps.ocp-c1.prod.psi.redhat.com/job/cnv-tests-runner-containerized/4034/

Anatw commented:
/verified

Anatw commented:
https://main-jenkins-csb-cnvqe.apps.ocp-c1.prod.psi.redhat.com/job/cnv-tests-runner-containerized/4038/

myakove commented:
> https://main-jenkins-csb-cnvqe.apps.ocp-c1.prod.psi.redhat.com/job/cnv-tests-runner-containerized/4031/

I get `This site can’t be reached` 
What is this link? how do I can see it?

Anatw commented:
> > https://main-jenkins-csb-cnvqe.apps.ocp-c1.prod.psi.redhat.com/job/cnv-tests-runner-containerized/4031/
> 
> I get `This site can’t be reached` What is this link? How can I open it?

This is an internal Red Hat link, I ran a module that uses this new PR and verified it.

myakove commented:
> > > https://main-jenkins-csb-cnvqe.apps.ocp-c1.prod.psi.redhat.com/job/cnv-tests-runner-containerized/4031/
> > 
> > I get `This site can’t be reached` What is this link? How can I open it?
> 
> This is an internal Red Hat link, I ran a module that uses this new PR and verified it.

I don't think it's ok to post here internal links.

Anatw commented:
/verified

Anatw commented:
I removed it

Anatw commented:
/verified

myakove reviewed (approved):
Internal links still exist in the PR comments.

Anatw commented:
> Internal links still exist in the PR comments.

I deleted all the references I have added. You need to delete your comments as well since some of the links are in them and I can't delete them from there myself.

Anatw commented:
/verified

Anatw commented:
/verified

rnetser commented:
/lgtm

Anatw commented:
/verified

Anatw commented:
/verified

rnetser commented:
@Anatw this pr has unresolved comments; comments must be resolved before merge

Anatw commented:
/verified

Anatw commented:
/cherry-pick v4.15

redhat-qe-bot2 commented:
Cherry-picked PR Add vlan and mtu parameters to OVNOverlayNetworkAttachmentDefinition into v4.15
